torch/csrc/jit/codegen/cuda/scheduler.cpp

```diff
@@ -57,159 +85,79 @@ bool scheduleFusion(Fusion* fusion, const at::ArrayRef<c10::IValue> inputs) {
-      TensorView* out_tv = output->as<TensorView>();
-      for (Val* inp : fusion->inputsOf(output)) {
-        if (inp->getValType().value() == ValType::TensorView)
-          inp->as<TensorView>()->computeAt(out_tv, -1);
-      }
-      out_tv->axis(0)->parallelize(ParallelType::BIDx);
-      out_tv->axis(1)->parallelize(ParallelType::Unroll);
-      out_tv->axis(2)->parallelize(ParallelType::TIDx);
+  for (auto output : fusion->outputs()) {
+    if (output->getValType() != ValType::TensorView)
+      continue;
+    TensorView* out_tv = output->as<TensorView>();
+    for (Val* inp : fusion->inputsOf(output)) {
+      if (inp->getValType().value() == ValType::TensorView)
+        inp->as<TensorView>()->computeAt(out_tv, -1);
     }
+    out_tv->axis(0)->parallelize(ParallelType::BIDx);
+    out_tv->axis(1)->parallelize(ParallelType::Unroll);
+    out_tv->axis(2)->parallelize(ParallelType::TIDx);
+  }
 
-    // Run through all values, unroll, and bind their axes
-    for (auto val : fusion->vals()) {
-      if (val->getValType().value() != ValType::TensorView ||
-          fusion->hasInput(val))
-        continue;
-      TensorView* tv = val->as<TensorView>();
-
-      // Disabling below as currently unrolling doesn't make a lot of sense as
-      // we don't extract global loads/reads out of intermediate logic.
-      //
-      // Below check should be true for all intermediates, but if one isn't
-      // hooked up right, skip it and hope for the best for now
-      //
-      // if (!disable_unroll && tv->nDims() == 3) {
-      //   tv->axis(-2)->parallelize(ParallelType::Unroll);
-      //   tv->axis(-1)->parallelize(ParallelType::TIDx);
-      // } else {
-      //   if (tv->nDims() == 2)
-      //     tv->axis(-1)->parallelize(ParallelType::TIDx);
-      // }
-    }
-    TensorView* out0 = fusion->outputs()[0]->as<TensorView>();
-    int ndim = (int)out0->nDims();
-    Val* numel = new Int(1);
-    for (int i = 0; i < ndim; i++) {
-      if (out0->axis(i)->isBlockDim()) {
-        numel = mul(numel, out0->axis(i)->rawExtent());
-      }
+  // Run through all values, unroll, and bind their axes
+  for (auto val : fusion->vals()) {
+    if (val->getValType().value() != ValType::TensorView ||
+        fusion->hasInput(val))
+      continue;
+    TensorView* tv = val->as<TensorView>();
+
+    // Disabling below as currently unrolling doesn't make a lot of sense as
+    // we don't extract global loads/reads out of intermediate logic.
+    //
+    // Below check should be true for all intermediates, but if one isn't
+    // hooked up right, skip it and hope for the best for now
+    //
+    // if (!disable_unroll && tv->nDims() == 3) {
+    //   tv->axis(-2)->parallelize(ParallelType::Unroll);
+    //   tv->axis(-1)->parallelize(ParallelType::TIDx);
+    // } else {
+    //   if (tv->nDims() == 2)
+    //     tv->axis(-1)->parallelize(ParallelType::TIDx);
+    // }
+  }
+  TensorView* out0 = fusion->outputs()[0]->as<TensorView>();
```

Comment:
code below `out0` should be cleaned as well. This is used in previously when scheduling is marked explicitly in transformation.